substrate-ros/robonomics/src/rosbag_player.rs

```diff
@@ -98,32 +96,22 @@ pub fn construct_player(path: String) -> Result<impl Future<Output=()>, Error> w
                 if start_msg_timestamp == 0 || msg_data.time < start_msg_timestamp {
                     start_msg_timestamp = msg_data.time;
                 }
-            }
-
+            },
             _ => ()
         }
     );
 
-    let mut players = vec![];
-    for ((storage_topic_name, topic_type), topics_ids) in storage_topics.iter() {
-        match topic_type.as_str() {
-            MSG_STRING_TYPE => players.push(RosbagPlayer::<std_msgs::String>::new(storage_topic_name, topics_ids.clone(), path.clone(), start_msg_timestamp.clone()).play()),
-            _ => println!("Unsupported topic type")
-        }
-    }
-    return Ok(future::join_all(players).map(|_| ()))
+    return Ok(build_players(storage_topics, path.clone(), start_msg_timestamp.clone()))
```

Comment:
It will be better to use macro this place according to spec. Instead of separate `construct_player` function.

Reply:
Using macro for this will makes generation and support of codebase much difficulty. Please see current implementation, it's simple to use and simple to support.